bidict/_base.py

```diff
@@ -486,7 +487,7 @@ def _init_from(self, other: MapOrIterItems[KT, VT]) -> None:
         self._fwdm.update(other)
         # If other is a bidict, use its existing backing inverse mapping, otherwise
         # other could be a generator that's now exhausted, so invert self._fwdm on the fly.
-        inv = other.inverse if isinstance(other, BidictBase) else inverted(self._fwdm)
+        inv: MapOrIterItems[t.Any, t.Any] = other.inverse if isinstance(other, BidictBase) else inverted(self._fwdm)
```

Comment:
The problem here is that `BidictBase` is both a valid `Mapping` and a valid `Iterable`, but its inferred type arguments are different in each case. The `isinstance` check narrows `other` to two possible specialized types of `BidictBase`.

If it's a `Mappable`, then it must be:
   `BidictBase[KT, VT]`
If it's an `Iterable`, then it must be:
   `BidictBase[tuple[KT, VT], Unknown]`

The problem is that only the latter type can be passed to the `self._invm.update` method.

Once again, I'm not sure why mypy doesn't complain about this. It may not attempt to infer type arguments when it's narrowing based on an `isinstance` call. It may also be due to the fact that mypy uses joins rather than unions in ternary statement evaluation, which loses some type information along the way.

In any case, I think pyright's behavior here is correct.

One way to type this without using an `Any` is as follows:
```python
        def is_bidict_base(obj: MapOrIterItems[KT, VT], x: type) -> TypeGuard[BidictBase[tuple[KT, VT], t.Any]]:
            return isinstance(obj, BidictBase)
        inv = other.inverse if is_bidict_base(other, BidictBase) else inverted(self._fwdm)
```

That's a little verbose though, so you may decide that the `MapOrIterItems[Any, Any]` is good enough.

Reply:
Thanks, @erictraut, good to see what that solution would look like.

I agree with you that it's a bit verbose though, and makes the code less maintainable. Also, `TypeGuard` requires Python 3.10+, and bidict currently supports 3.7+.

This looks like another case where something like a `# pyright: ignore` inline comment would be a preferable option for libraries like bidict. Is there any other equally expedient way to appease pyright here, without resorting to the `MapOrIterItems[Any, Any]`?

Reply:
For now I'm going with a `pyright: ignore` here. In the meantime, I'll leave this thread "unresolved" in case @JelleZijlstra wants to follow up on whether mypy should also be flagging this after all, and possibly link a newly-filed bug in mypy.

Reply:
I don't think this is worth changing in mypy.

Reply:
Thanks for confirming. Marking this resolved.